test/AlternativeSpec.hs

```diff
@@ -38,3 +45,11 @@ spec = parallel $ do
         `shouldBe` Just (["salabim"], ())
       (run . runTraceList . runNonDetMaybe) failtrace'
         `shouldBe` (["sim", "salabim"], Just ())
+
+  describe "runNonDet" $ do
+    it "should have terrible semantics when <|> is used within\
```

Comment:
excellent test name

Reply:
could we also write an "expect failure" for what it should be, just to document that?

Reply:
Good idea. Simply via `shouldNotBe`, or is there something better?

Reply:
There's nothing better :( but god would it be nice if there were.